File: URI/C++/3037.cpp
#include<iostream>
using namespace std;

int main(){
    int N, maria, joao, ponto, dist;

    cin >> N;
    for(int i=0; i<N; i++){
        maria = 0, joao = 0;
        for(int h=0;h<3;h++){
            cin >> ponto >> dist;
            joao += ponto*dist;
        }
        for(int h=0;h<3;h++){
            cin >> ponto >> dist;
            maria += ponto*dist;
        }
        if(maria > joao)
            cout << "MARIA\n";
        else
            cout << "JOAO\n";
//não em uma opção pra empate
    }

    return 0;
}

File: URI/Python/3037.py
N = int(input())

for i in range(N):
    maria = 0
    joao = 0
    for i in range(3):
        ponto, dist = list(map(int,input().split()))
        joao += ponto*dist
    for i in range(3):
        ponto, dist = list(map(int,input().split()))
        maria += ponto*dist
    if maria > joao:
        print("MARIA")
    else:
        print("JOAO")
